packages/main/src/GridWalker.ts

```diff
@@ -0,0 +1,115 @@
+/**
+ * The GridWalker object represents the elements of a two-dimensional grid system and a position within them.
+ *
+ * @class
+ * @private
+ */
+class GridWalker {
+	grid: unknown[][];
+	rowPos: number;
+	colPos: number;
+	pageSize: number;
+
+	constructor(grid: unknown[][] = [[]], pageSize = 2, rowPos = 1, colPos = 0) {
+		this.grid = grid;
+		this.pageSize = pageSize;
+		this.rowPos = rowPos;
+		this.colPos = colPos;
+	}
+
+	setGrid(grid: unknown[][], initColPos = false) {
+		this.grid = grid;
+		this._updateRowPos();
+		if (initColPos) {
+			this.colPos = 0;
+		} else {
+			this._updateColPos();
+		}
+	}
+
+	left() {
+		this.colPos = Math.max(this.colPos - 1, 0);
+	}
+
+	right() {
+		this.colPos = Math.min(this.colPos + 1, this.grid[this.rowPos].length - 1);
+	}
+
+	up() {
+		this.rowPos = Math.max(this.rowPos - 1, 0);
+		this._updateColPos();
```

Comment:
Just for my understanding, why are the col positions updated here, because they should not change when calling up/down, no?